apis/capabilities/v1beta1/application_types.go

```diff
@@ -0,0 +1,140 @@
+/*
+Copyright 2020 Red Hat.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1beta1
+
+import (
+	"github.com/3scale/3scale-operator/pkg/common"
+	"github.com/go-logr/logr"
+	"github.com/google/go-cmp/cmp"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"reflect"
+)
+
+const (
+	ApplicationReadyConditionType common.ConditionType = "Ready"
+)
+
+// EDIT THIS FILE!  THIS IS SCAFFOLDING FOR YOU TO OWN!
+// NOTE: json tags are required.  Any new fields you add must have json tags for the fields to be serialized.
+
+// ApplicationSpec defines the desired state of Application
+type ApplicationSpec struct {
+	// INSERT ADDITIONAL SPEC FIELDS - desired state of cluster
```

Comment:
missing provider account 

```
    // ProviderAccountRef references account provider credentials                        
    // +optional                                                                         
    ProviderAccountRef *corev1.LocalObjectReference `json:"providerAccountRef,omitempty"`

```

Reply:
Yeah, the application uses the developer account for its reference. Not sure where I should document that at the moment, could you point me in the right direction?

I've added a tiny bit of logic to the externalRefCheck() that will check if both resources have the same providerAccountHost if not then it should return an error.

Reply:
Guessing this should be documented in the `doc` directory under `application-reference.md` I will make a start on it.

Reply:
added documentation
https://github.com/3scale/3scale-operator/pull/778/files#diff-539a57d01480bffad575ce52c6aac9277d2495037aee29b0309811d92e0f6855

Reply:
With the addition of the docs and the checkExternalResources() check https://github.com/3scale/3scale-operator/pull/778/commits/4711eb55d6d7be3e7d39f51d6f7867e2fa3600d3#diff-067ba2ea2f14e8822e886fc126454337bc5e488fe598f1a6f2f6ad536c572dabR259 think this is resolved.